cmd/prometheus/main.go

```diff
@@ -1546,7 +1550,8 @@ func (opts tsdbOptions) ToTSDBOptions() tsdb.Options {
 		RetentionDuration:              int64(time.Duration(opts.RetentionDuration) / time.Millisecond),
 		MaxBytes:                       int64(opts.MaxBytes),
 		NoLockfile:                     opts.NoLockfile,
-		AllowOverlappingBlocks:         opts.AllowOverlappingBlocks,
+		AllowOverlappingCompaction:     opts.AllowOverlappingBlocks,
+		AllowOverlappingQueries:        opts.AllowOverlappingBlocks,
```

Comment:
Imagina you turn on and then off out of order. You local filesystem may still have overlapping blocks because you had out of order on and you want to query them. In that scenario you want to keep `AllowOverlappingQueries` set to true.

For compaction is different. When OOO is true the actual implementation expectes to generate overlapping compacted blocks if ooo samples were received. However if OOO is false we're never expecting overlapping compacted blocks.

Reply:
This switch was added to also be helpful to downstream users where you want vertical queries but not vertical compaction. With the proposal of always allowing overlapping blocks in Prometheus, we will be able to get rid of the switch for overlapping queries and only have it for compactions later as a follow up work.